compiler/rustc_middle/src/mir/interpret/allocation/provenance_map.rs

```diff
@@ -179,37 +203,27 @@ impl<Prov: Provenance> ProvenanceMap<Prov> {
 
         // We need to handle clearing the provenance from parts of a pointer.
         if first < start {
-            if !Prov::OFFSET_IS_ADDR {
-                // We can't split up the provenance into less than a pointer.
-                return Err(AllocError::OverwritePartialPointer(first));
```

Comment:
I guess one way to feature gate it would be to change all of the assert or error cases to instead raise a feature gate error

Reply:
That seems odd as it is a dynamic check, performed during const-eval execution. We could have crate A do something weird inside its unsafe code that then leads to a feature gate error in a constant declared in crate B -- and in all crates using that constant.